### PR TITLE
feat: payment request page data persist

### LINKED ISSUE #1813: [settings - UPE] Stripe settings persistence - Express checkouts page
Part of #1717.

Add settings data API integration to the "Express checkouts" page.

### PR DESCRIPTION
**Pull request by frosso**
# Changes proposed in this Pull Request:
Fixes https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1813

Allows persisting the data on the "Express checkouts" page.

I moved the files from `client/settings/payment-requests/*` to `client/entrypoints/payment-request-settings/*`, and ensured that the entrypoint `payment_requests_customizer` matches the file name more closely (so that it's easier to find).

**PLEASE NOTE:** the button preview is still not working, because the account keys are missing - separate ticket for it: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1846

# Testing instructions
- Ensure you have the `_wcstripe_feature_upe_settings` flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&area=payment_requests
- Change some of the values
- Click the "Save changes" button
- Data is persisted
- Refresh the page
- The same values you selected are now displayed

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -26,8 +26,8 @@ module.exports = {
 	},
 	entry: {
 		index: './client/blocks/index.js',
-		payment_requests_customizer:
-			'./client/settings/payment-requests/index.js',
+		payment_requests_settings:
```
Comment (frosso) [inline webpack.config.js:29]: - Moved the page to the `entrypoints` directory (see pc4etw-jM-p2 for background)
- Ensured the name of the directory matches the name of the entrypoint more closely.

File: includes/admin/stripe-settings.php [hunk 2]
```diff
@@ -238,6 +250,20 @@
 	}
 	// Insert UPE options below the 'description' setting.
 	$stripe_settings = array_merge( array_splice( $stripe_settings, 0, array_search( 'description', array_keys( $stripe_settings ), true ) + 1 ), $upe_settings, $stripe_settings );
+
+	// in the new settings, "checkout" is going to be enabled by default (if it is a new WCStripe installation).
+	$stripe_settings['payment_request_button_locations']['default'][] = 'checkout';
```
Comment (frosso) [inline includes/admin/stripe-settings.php:255]: I created a migration ticket for this: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1845

File: includes/admin/class-wc-stripe-payment-requests-controller.php [hunk 3]
```diff
@@ -19,46 +19,41 @@ public function __construct() {
...
-	 * @param WC_Stripe_Payment_Gateway $gateway the Stripe gateway.
 	 */
-	public function admin_options( WC_Stripe_Payment_Gateway $gateway ) {
+	public function admin_options() {
 		global $hide_save_button;
 		$hide_save_button = true;
 		echo '<h2>' . __( 'Customize express checkouts', 'woocommerce-gateway-stripe' );
 		wc_back_link( __( 'Return to Stripe', 'woocommerce-gateway-stripe' ), admin_url( 'admin.php?page=wc-settings&tab=checkout&section=stripe' ) );
 		echo '</h2>';
-		echo '<div class="wrap"><div id="wc_stripe-payment-requests_customizer_container"></div></div>';
```
Comment (frosso) [inline includes/admin/class-wc-stripe-payment-requests-controller.php:62]: Ensured that the name is more consistent with the other ones

File: client/components/loadable-settings-section/index.js [hunk 4]
```diff
@@ -1,24 +0,0 @@
-/**
```
Comment (frosso) [inline client/components/loadable-settings-section/index.js:1]: This was duplicated

**Comment by tommyshellberg** [comment]
Just a note that this will require a fix to the `useSettings()` hook because `isLoading` is stuck at `true` which prevents the component from loading.

With a hotfix applied, it's working 🎊 

additionally, I actually see the Preview buttons using Chrome + my Jurassic.tube URL:

Screen Shot 2021-09-15 at 8 27 05 PM

File: client/entrypoints/payment-request-settings/payment-request-section.js [hunk 5]
```diff
@@ -8,18 +8,16 @@ import interpolateComponents from 'interpolate-components';
 import { Elements } from '@stripe/react-stripe-js';
```
Comment (tommyshellberg) [inline client/entrypoints/payment-request-settings/payment-request-section.js:8]: `useState` can be removed as an import, no longer being used.

Comment (frosso) [reply client/entrypoints/payment-request-settings/payment-request-section.js:8]: Ah thanks! Done in 865df282

File: client/entrypoints/payment-request-settings/payment-request-page.js [hunk 6]
```diff
@@ -49,12 +49,12 @@ const Description = () => (
...
-const PaymentRequestsSettings = () => {
+const PaymentRequestsPage = () => {
 	return (
 		<SettingsLayout>
 			<SettingsSection Description={ Description }>
 				<LoadableSettingsSection numLines={ 30 }>
-					<PaymentRequestsCustomizer />
+					<PaymentRequestsSection />
 				</LoadableSettingsSection>
 			</SettingsSection>
```
Comment (tommyshellberg) [inline client/entrypoints/payment-request-settings/payment-request-page.js:60]: extra empty line(that's from my PR 😬 )

Comment (frosso) [reply client/entrypoints/payment-request-settings/payment-request-page.js:60]: I don't mind it tho - it kinda separates the `SettingsSection` from the `SaveSettingsSection` :D would you like me to remove it anyways?

File: includes/admin/class-wc-rest-stripe-settings-controller.php [hunk 7]
```diff
@@ -86,6 +129,7 @@ public function get_settings() {
 	 */
 	public function update_settings( WP_REST_Request $request ) {
 		$this->update_is_payment_request_enabled( $request );
+		$this->update_payment_request_settings( $request );
```
Comment (tommyshellberg) [inline includes/admin/class-wc-rest-stripe-settings-controller.php:132]: One thing I noticed is that the fetch call to this endpoint will catch exceptions but there aren't any exceptions thrown in this class. Are there conceivable paths where we'd want to throw an exception(an update fails?) or are we mainly just catching network errors?

File: includes/admin/class-wc-stripe-payment-requests-controller.php [hunk 8]
```diff
@@ -19,46 +19,41 @@ public function __construct() {
...
+		wp_enqueue_script( 'wc-stripe-payment-request-settings' );
 
 		wp_register_style(
-			'wc_stripe-payment-requests_customizer',
-			plugins_url( 'build/payment_requests_customizer.css', WC_STRIPE_MAIN_FILE ),
-			[],
-			$script_asset['version']
+			'wc-stripe-payment-request-settings',
+			plugins_url( 'build/payment_requests_settings.css', WC_STRIPE_MAIN_FILE ),
+			[ 'wc-components' ],
```
Comment (tommyshellberg) [inline includes/admin/class-wc-stripe-payment-requests-controller.php:41]: oops, looks like I missed this dependency in my original PR. What tipped you off that you needed it?

File: includes/admin/class-wc-rest-stripe-settings-controller.php (hunk 7 above)
Comment (frosso) [reply includes/admin/class-wc-rest-stripe-settings-controller.php:132]: The `POST` request to `update_settings` could fail when the validation on the input fails (through `validate_callback`) - is that the endpoint you're referring to?
In that case, an exception can be thrown by the validators.

File: includes/admin/class-wc-stripe-payment-requests-controller.php (hunk 8 above)
Comment (frosso) [reply includes/admin/class-wc-stripe-payment-requests-controller.php:41]: Nothing much - just covering our bases. I think other scripts on the page already queue them, but I just wanted to explicitly declare them here as well

**Comment by frosso** [comment]
> Just a note that this will require a fix to the `useSettings()` hook because `isLoading` is stuck at `true` which prevents the component from loading.

Should be resolved, now that https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1864 is merged! 🎉

File: includes/admin/class-wc-rest-stripe-settings-controller.php (hunk 7 above)
Comment (tommyshellberg) [reply includes/admin/class-wc-rest-stripe-settings-controller.php:132]: Sounds good 👍

**Review by tommyshellberg — APPROVED** [review]
Nice work! Working now after the `useSettings()` fix 🥳

**Pull request merged**